code/modules/mob/living/simple_animal/friendly/farm_animals.dm

```diff
@@ -317,16 +317,21 @@
 
 
 /obj/item/udder
-	name = "udder"
+        name = "udder"
+        var/milktype = "milk"
+
+/obj/item/udder/goat
+        name = "goat udder"
+        milktype = "goatmilk"
 
 /obj/item/udder/Initialize()
-	create_reagents(50)
-	reagents.add_reagent("milk", 20)
-	. = ..()
+        create_reagents(50)
+        reagents.add_reagent("[milktype]", 20)
+        . = ..()

```

Comment:
this has 0 effect currently but put this as the first line under Initialize in case someone adds early returns.